loaders/android/bootstrap.c.in

```diff
@@ -62,11 +62,23 @@ void Java_@SYS_PACKAGE_UNDERSCORE@_@SYS_APPNAME@_nativeEvent(JNIEnv* e, jobject
 // JNI Hooks and Global Objects
 static jobject globalObj=NULL;
 static JavaVM* s_vm = NULL;
+static const char* app_directory_files = NULL;
+static const char* app_code_path = NULL;
 
-void Java_@SYS_PACKAGE_UNDERSCORE@_@SYS_APPNAME@_nativeInstanceInit(JNIEnv* env, jobject thiz){
-  globalObj = (*env)->NewGlobalRef(env,thiz); 
+void Java_@SYS_PACKAGE_UNDERSCORE@_@SYS_APPNAME@_nativeInstanceInit(JNIEnv* env, jobject thiz, jstring codePath, jstring directoryFiles){
+
+  globalObj = (*env)->NewGlobalRef(env,thiz);
+  app_directory_files  = strdup((*env)->GetStringUTFChars(env, directoryFiles, 0));
+  (*env)->ReleaseStringUTFChars(env, directoryFiles, NULL);
+  app_code_path = strdup((*env)->GetStringUTFChars(env, codePath, 0));
+  (*env)->ReleaseStringUTFChars(env, codePath, NULL);
 }
 
+char* android_getFilesDir() { return (char*) app_directory_files; }
+char* android_getPackageCodePath() { return (char*) app_code_path; }
+
+char* android_getFilesDir_info_get() { return android_getFilesDir(); }
```

Comment:
Sorry why is this needed - it does the exact same thing as android_getFilesDir(), which is defined 2 lines above?